Library/Homebrew/build_environment.rb

```diff
@@ -55,22 +55,23 @@ def env(*settings)
   ].freeze
   private_constant :KEYS
 
-  sig { params(env: T.untyped).returns(T::Array[String]) }
+  sig { params(env: T::Hash[String, T.nilable(T.any(String, Pathname))]).returns(T::Array[String]) }
   def self.keys(env)
     KEYS & env.keys
   end
 
-  sig { params(env: T.untyped, f: IO).void }
+  sig { params(env: T::Hash[String, T.nilable(T.any(String, Pathname))], f: IO).void }
   def self.dump(env, f = $stdout)
```

Comment:
Does the value being `nil` make this effectively a no-op?

Reply:
It would still print, it just wouldn't resolve a filename for nil

Reply:
Ah gotcha.